portability-api/src/main/java/org/datatransferproject/api/action/transfer/StartTransferJobAction.java

```diff
@@ -54,66 +42,40 @@ public TransferJob handle(StartTransferJob startTransferJob) {
     UUID jobId = decodeJobId(id);
     PortabilityJob job = jobStore.findJob(jobId);
 
-    if (!startTransferJob.isAuthDataEncrypted()) {
-      // TODO: move creds encryption to the client, and pass encrypted creds to this action.
-      // Update this job with credentials encrypted with a public key, e.g. for a specific transfer
-      // worker instance
-      job =
-          encryptAndUpdateJobWithCredentials(
-              jobId,
-              job,
-              startTransferJob.getExportAuthData(),
-              startTransferJob.getImportAuthData());
-    }
-
-    // TODO(#553): This list of nulls should be cleaned up when we refactor TransferJob.
-    return new TransferJob(id, job.exportService(), job.importService(), job.transferDataType(),
-            null, null, null, null, null, null);
+    job = updateJobWithCredentials(jobId, job, startTransferJob.getEncryptedAuthData());
```

Comment:
I think we should be OK since this is covered in JWE. If we move encryption of the initial auth data for legacy auth providers, we should be able to eliminate all of the encryption from the server.